source/common/grpc/google_grpc_utils.cc

```diff
@@ -127,7 +127,7 @@ GoogleGrpcUtils::channelArgsFromConfig(const envoy::config::core::v3::GrpcServic
       break;
     }
     default:
-      NOT_REACHED_GCOVR_EXCL_LINE;
+      throw EnvoyException("Oneof name not set");
```

Comment:
What do you think about using protoc-gen-validate https://github.com/envoyproxy/protoc-gen-validate to ensure this condition before it throws? 
(FWIW I usually see NOT_REACHED_GCOVR_EXCL_LINE, but only when it's impossible to actually hit the line. Which isn't true unless a constraint holds.)

Reply:
Yeah, that's definitely a cleaner solution. Updated to use protoc-gen-validate and rolled back changes to source/common/grpc/google_grpc_utils.cc in the latest commit.